odin/src/main/scala/org/clulab/odin/impl/Quantifier.scala

```diff
@@ -0,0 +1,37 @@
+package org.clulab.odin.impl
+
+/**
+  * Represents a regex quantifier
+  */
+trait Quantifier {
```

Comment:
Since this is specific to `ArgumentPattern`, maybe rename to `ArgumentQuantifier` or something similar?